File: Javascript/ConvertSortedArrayToBinarySearchTree.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0 : val
    this.left = (left === undefined) ? null : left
    this.right = (right === undefined) ? null : right
}

var sortedArrayToBST = function(nums){
    const start = 0
    const end = nums.length-1

    function helper(start,end){
        if(end-start === 0){
            return new TreeNode(nums[start])
        }
        else if(end-start < 0){
            return null
        }

        const middle = Math.floor((start + end)/2)
        const middleNode = new TreeNode(nums[middle])

        middleNode.left = helper(start,middle-1)
        middleNode.right = helper(middle+1,end)

        return middleNode
    }

    return helper(start,end)
}

File: Python/ConvertSortedArrayToBinarySearchTree.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


class Solution:

    def sortedArrayToBST(nums):

        start = 0
        end = len(nums)-1

        def helper(start,end):
            if end-start == 0:
                return TreeNode(nums[start])
            elif end-start < 0:
                return None
            
            middle = (start + end) // 2
            middleNode = TreeNode(nums[middle])

            middleNode.left = helper(start,middle-1)
            middleNode.right = helper(middle+1,end)

            return middleNode
        
        return helper(start,end)